infra/vsts_agent_ubuntu_20_04_startup.sh

```diff
@@ -0,0 +1,198 @@
+
+#Start docker daemon
+systemctl enable docker
+
+## Install the VSTS agent
+groupadd --gid 3000 vsts
+useradd \
+  --create-home \
+  --gid 3000 \
+  --shell /bin/bash \
+  --uid 3000 \
+  vsts
+#add docker group to user
+usermod -aG docker vsts
+
+su --login vsts <<'AGENT_SETUP'
+set -euo pipefail
+
+VSTS_ACCOUNT=${vsts_account}
+VSTS_POOL=${vsts_pool}
+VSTS_TOKEN=${vsts_token}
+
+mkdir -p ~/agent
+cd ~/agent
+echo 'assignment=default' > .capabilities
+
+echo Determining matching VSTS agent...
+VSTS_AGENT_RESPONSE=$(curl -sSfL \
+  -u "user:$VSTS_TOKEN" \
+  -H 'Accept:application/json;api-version=3.0-preview' \
+  "https://$VSTS_ACCOUNT.visualstudio.com/_apis/distributedtask/packages/agent?platform=linux-x64")
+
+VSTS_AGENT_URL=$(echo "$VSTS_AGENT_RESPONSE" \
+  | jq -r '.value | map([.version.major,.version.minor,.version.patch,.downloadUrl]) | sort | .[length-1] | .[3]')
+
+if [ -z "$VSTS_AGENT_URL" -o "$VSTS_AGENT_URL" == "null" ]; then
+  echo 1>&2 error: could not determine a matching VSTS agent - check that account \'$VSTS_ACCOUNT\' is correct and the token is valid for that account
+  exit 1
+fi
+
+echo Downloading and installing VSTS agent...
+curl -sSfL "$VSTS_AGENT_URL" | tar -xz --no-same-owner
+
+set +u
+source ./env.sh
+set -u
+
+./config.sh \
+  --acceptTeeEula \
+  --agent "$(hostname)" \
+  --auth PAT \
+  --pool "$VSTS_POOL" \
+  --replace \
+  --token "$VSTS_TOKEN" \
+  --unattended \
+  --url "https://$VSTS_ACCOUNT.visualstudio.com"
+AGENT_SETUP
+
+## Hardening
+
+chown --recursive root:root /home/vsts/agent/{*.sh,bin,externals}
+
+## Install Nix
+
+# This needs to run inside of a user with sudo access
+echo "vsts ALL=(ALL:ALL) NOPASSWD:ALL" > /etc/sudoers.d/nix_installation
+su --command "sh <(curl -sSfL https://nixos.org/nix/install) --daemon" --login vsts
+rm /etc/sudoers.d/nix_installation
+
+# Note: the "hydra.da-int.net" string is now part of the name of the key for
+# legacy reasons; it bears no relation to the DNS hostname of the current
+# cache.
+cat <<NIX_CONF > /etc/nix/nix.conf
+binary-cache-public-keys = hydra.da-int.net-2:91tXuJGf/ExbAz7IWsMsxQ5FsO6lG/EGM5QVt+xhZu0= hydra.da-int.net-1:6Oy2+KYvI7xkAOg0gJisD7Nz/6m8CmyKMbWfSKUe03g= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.nixos.org-1:CNHJZBh9K4tP3EKF6FkkgeVYsS3ohTl+oS0Qa8bezVs=
+binary-caches = https://nix-cache.da-ext.net https://cache.nixos.org
+build-users-group = nixbld
+cores = 1
+max-jobs = 0
+sandbox = relaxed
+NIX_CONF
+
+systemctl restart nix-daemon
+
+# Warm up local caches by building dev-env and current daml main
+# This is allowed to fail, as we still want to have CI machines
+# around, even when their caches are only warmed up halfway
+su --login vsts <<'CACHE_WARMUP'
+# user-wide bazel disk cache override
+echo "build:linux --disk_cache=~/.bazel-cache" > ~/.bazelrc
+
+# clone and build
+(
+  git clone https://github.com/digital-asset/daml
+  cd daml
+  git checkout upgrade-linux-nodes-to-20.04
```

Comment:
This is not required once this PR is merged into main, so I'll be redeploying the cluster without this line. Note that this only runs when we rebuild a machine, so a bit of a lag here is acceptable.